codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -156,17 +156,20 @@ public void addContent(ByteBuf buffer, boolean last)
                     throw new IOException("Out of size: " + (size + localsize) +
                             " > " + definedSize);
                 }
-                int written = 0;
                 if (file == null) {
                     file = tempFile();
                 }
                 if (fileChannel == null) {
                     RandomAccessFile accessFile = new RandomAccessFile(file, "rw");
                     fileChannel = accessFile.getChannel();
                 }
-                buffer.getBytes(buffer.readerIndex(), fileChannel, fileChannel.position(), localsize);
+                long currentPosition = fileChannel.position();
+                int written = buffer.getBytes(buffer.readerIndex(), fileChannel, currentPosition, localsize);
+                if (written > 0) {
+                    fileChannel.position(currentPosition + written);
+                    buffer.readerIndex(buffer.readerIndex() + written);
+                }
```

Comment:
I think this may actually need to be a loop since it's not guaranteed to write all of the bytes in a single invocation

Reply:
@amizurov I think @njhill is right here... can you adjust this please ?

Reply:
Sure, according java docs it depends on implementation.

Reply:
instead of `if (written > 0)`, you might better inverse the test to `if (written < 0)` which stands for end of file/channel, therefore giving a break of the `while` loop ?

Reply:
or `if (written <= 0)` since the javadoc does not specify negative result (as in standard Java interaction with file or stream). The idea is to have a clear "break" code, and therefore to not rely only on while condition.

Reply:
@amizurov ^^

Reply:
I changed code according suggestion but i'm not sure is it ok if we break from loop but possible not all data is written or maybe i'm confused :)

Reply:
@amizurov I think we should only break on `< 0` as `0` is valid.

Reply:
Ok, also moved change position and buffer reader index from loop